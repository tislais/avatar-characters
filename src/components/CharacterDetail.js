import React, { Component } from 'react';
import { fetchCharacterById } from '../services/avatarApi';

export default class CharacterDetail extends Component {
  state = {
    character: []
  };

  async componentDidMount() {
    const requestedCharacter = this.props;
    
    try {
      const character = await fetchCharacterById(requestedCharacter.match.params.id);
      this.setState({ character: character });
    } catch (err) {
      console.log(err);
    }
  }

  render() {
    const { character } = this.state;

    return (
      <>
        <h1 className="text-4xl">{character.name}</h1>
        <img src={character.photo} alt={character.name} />
        <p>Gender: {character.gender}</p>
        <p>Hair Color: {character.hair}</p>
        <p>Position: {character.position}</p>
        <p>Affiliation: {character.affiliation}</p>
        <p>Allies: {character.allies}</p>
        <p>Enemies: {character.enemies}</p>
      </>
    )
  }
}