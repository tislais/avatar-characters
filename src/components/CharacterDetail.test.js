import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';

describe('Character detail component', () => {
  it('renders Character', () => {
    const { asFragment } = render(<CharacterDetail
      name="46th Earth King"
      allies={["Royal Earthbender Guards"]}
      enemies={["Chin"]}
      photo="https://vignette.wikia.nocookie.net/avatar/images/5/51/46th_Earth_King.png/revision/latest?cb=20130627160441" />);
    expect(asFragment()).toMatchSnapshot();
  });
});