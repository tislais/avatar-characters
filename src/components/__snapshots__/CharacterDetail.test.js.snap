// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character detail component renders Character 1`] = `
<DocumentFragment>
  <h1
    class="text-4xl"
  />
  <img />
  <p>
    Gender: 
  </p>
  <p>
    Hair Color: 
  </p>
  <p>
    Position: 
  </p>
  <p>
    Affiliation: 
  </p>
  <p>
    Allies: 
  </p>
  <p>
    Enemies: 
  </p>
</DocumentFragment>
`;
